src/app.js

```diff
@@ -112,8 +116,35 @@ const PerfectyPush = (() => {
     await Registration.check(Storage.userId(), Storage.optedOut())
   }
 
+  const register = async () => {
+    Options.askPermissionsDirectly = true
```

Comment:
We should not alter user options but instead adjust the existing code to support explicit registrations called from JS. I will refactor the current code in Perfecty in order to support what you intend to do in a separate PR.